In [18]:
# # Auxiliares ao desenvolvimento, remover uma vez que o código está funcional
%load_ext autoreload
%autoreload 2
%reload_ext autoreload

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: 

In [19]:
import os
from maquinas.dados import Dados

dados = Dados.ler_dados(nome_arquivo='../dados/dados.txt', separador=",")

dados

<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'


{'Uan': '240',
 'Ian': '10',
 'Ra': '2',
 'Rf': '120',
 'Rreos': '0-100',
 'Ke': '0.8',
 'Kt': '0.8',
 'J': '0.02',
 'RPM': '1800',
 'Pn': '2.5',
 'n': '85',
 'T_L': '8-15'}

In [20]:
from maquinas.motor_cc_shunt import MotorCCShunt
from maquinas.dados import Dados


motor = MotorCCShunt(dados=dados)

motor


Objeto MotorCCShunt criado... Parâmetros:
	 Uan : 	 240
	 Ian : 	 10
	 Ra : 	 2
	 Rf : 	 120
	 Rreos : 	 0-100
	 Ke : 	 0.8
	 Kt : 	 0.8
	 J : 	 0.02
	 RPM : 	 1800
	 Pn : 	 2.5
	 n : 	 85
	 T_L : 	 8-15


In [21]:
motor.calcula_fem()
# motor.calcula_w_rads()


220.0

In [ ]:
motor.calcula_torque()
motor.calcula_fluxo_tensao()
motor.calcula_fluxo_torque()
motor.calcula_const_magnetica()
motor.calcula_corrente_campo()
motor.calcula_corrente_armadura()
# motor.calcula_velocidade()
# motor.calcula_potencias()

AttributeError: 'MotorCCShunt' object has no attribute 'calcula_velocidade'

# Fluxo de trabalho geral

# Fluxo de trabalho detalhado

1. Criar estruturas para leitura e escrita de arquivos
   1. Essas estruturas devem ler os dados de forma padronizada e retornar um dicionário para posterior uso
      1. O dicionário pode conter campos para comentário
2. Criar uma classe que consuma os dados lidos dos arquivos
3. 

# Condições de operação

## Condição 1: Operação nominal (240 V, sem reostato, carga nominal);
## Condição 2: Subtensão (200 V, carga nominal);
## Condição 3: Enfraquecimento de campo (reostato = 50 Ω);
## Condição 4: Baixa carga (torque reduzido);
## Condição 5: Condição crítica (baixa tensão e alto torque).

In [ ]:
#